Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift

```diff
@@ -679,7 +679,9 @@ class HTTPUpgradeTestCase: XCTestCase {
         try server.eventLoop.submit {
             upgrader.unblockUpgrade()
         }.wait()
-        client.close(promise: nil)
+        try client.eventLoop.scheduleTask(in: .milliseconds(10)) {
```

Comment:
@Lukasa can you expand why it helps to wait 10ms here?

Reply:
10ms isn't a magic number, we just need to wait *long enough*.

The problem is that this close is racing with the `unblockUpgrade` operation above. It's possible that the close wins, and the connection is torn down before the client receives the 101 response. This wait tries to ensure that the client sees the 101 before connection termination.

Reply:
We can do this another way by writing a custom accumulator handler that behaves differently to the one we're currently using, if you'd prefer.

Reply:
@Lukasa sorry I think I'd prefer that 😬

Reply:
Sure thing, I'll rework this.